### DISCUSSION TITLE
New gases, Internal pressure of the submarine, Temperature.

### DISCUSSION THREAD

**Discussion by Lomikk** [FakeFishGames/Barotrauma#7235] [Ideas and suggestions]
Hello, I want to tell you a few of my ideas, most of which can be implemented through the mechanics that are already in the game.
1 ) **New gases and an oxygen generator**, now there would be not only oxygen in the air, but also other harmful impurities (co₂) (Hg) (H), etc.  _About how they will appear on the boat in more detail in paragraph 2_. To get rid of them, you would need to buy filters for an oxygen generator or use a breathing mask, inhaling contaminated air would be harmful to health (also, ballast flora could emit toxic gases)

![image](https://user-images.githubusercontent.com/92258604/140971602-810148e7-2c2d-45a8-b853-f10ef1f914d3.png)

-If gases will be added , I consider it necessary to add an impurity analyser in the air and a Geiger counter.

2) **External factors affect the contents of containers**. You already have mechanics because of which potassium(K) and sodium(Na) explode upon contact with water, you could give these elements the opportunity to pollute the air, (Hg) could be found in mines and it would evaporate in the warm atmosphere of a submarine, (H) hydrogen could be obtained by exploding a cutter cylinder, (co₂) and carbon dioxide would simply accumulate because of the crew existence within submarine.

![image](https://user-images.githubusercontent.com/92258604/140972073-e86b6423-36d2-49a2-9b2a-5157c0ca089e.png)

3) **Internal pressure of the submarine**. If the oxygen generator malfunctioned or the oxygen tank was destroyed, the pressure in the rooms would increase, or vice versa, if there were holes, pressure would fall.

![image](https://user-images.githubusercontent.com/92258604/140972488-c4106772-4e93-455f-a6b5-a4b284d360eb.png)

4) **Temperature**. During a fire, compressed gas cylinders could explode or start flying around the room (or they can simply be fired at with the gun when they are lying on the floor)

![image](https://user-images.githubusercontent.com/92258604/140972899-2c2e4398-8bed-400a-b02b-96e75bd1a554.png)

5) **Water could pass electricity if the cable box malfunctions.** Those who wear special equipment will be able to avoid damage from electricity.

![image](https://user-images.githubusercontent.com/92258604/140973402-c2c55339-9969-4be0-96a1-26a6c67ce532.png)

**Comment by MaaiG** [comment]
I agree. At least these possibilities may be optional.
- Harmful gases, which are heavier, shall be applied to the ground (filling hull from bottom). Those that are easier to climb to the ceiling (fill the hull from above). It also affects the breath of the character and his poisoning (if your head is in a cloud of gas).
- Smoke from combustion may be black or white depending on the source of the combustion. When doors are opened, the smoke flows into other shells. To get rid of it - it is necessary to ventilate premises.
- Electricity in water is a good idea, can make simple repairs very difficult and turn it into a real quest (if the submarine is designed correctly). But we have to adjust the wear pattern.
The complexity of repairs will increase and will require a reduction in their unprofitable number.
In fact, it would be right if the boat’s systems didn’t wear out when they were Off or if they not exposed to external negative influences, as they are now.
Systems shall only deteriorate when the devices are in operation or damaged. My submarines have a lot of systems that only turn on when necessary, but they’re constantly worn out at rest. It’s not logical that we have to do all the repairs on a big boat, repair the life pods and the shuttles that weren’t even used.
"At the moment of the submarine’s death at critical depth, you run into the escape pod, open the airlock, jump into it, and... 
in the escape pod, you need to repair: electrics, controls, ballast pumps, engines, oxygen, and... 
you’re dead!"

**Reply by Lomikk** [reply]
> Harmful gases, which are heavier, shall be applied to the ground (filling hull from bottom). Those that are easier to climb to the ceiling (fill the hull from above). It also affects the breath of the character and his poisoning (if your head is in a cloud of gas).


this is a good idea, there is such a thing in the game oxygen not included.
In fact, I would really like to see a similar system in Barotrauma, it would be fun if some materials could evaporate from leaky containers.

![image](https://user-images.githubusercontent.com/92258604/141159988-cedf09bf-7f5c-4561-8923-289d8c007325.png)

**Reply by MaaiG** [reply]
> it would be fun if some materials could evaporate from leaky containers.

Cool! Slow and invisible release of toxins from an externally damaged container! Ultimately, the classic case - spilled acid can burn several decks ))
